src/generate.rs

```diff
@@ -51,65 +53,75 @@ impl GenMode {
-    match attr {
-        // `#[get = "pub"]` or `#[set = "pub"]`
-        Some(Meta::NameValue(MetaNameValue {
-            lit: Lit::Str(ref s),
-            path,
-            ..
-        })) => {
-            if path.is_ident(meta_name) {
-                s.value().split(' ').find(|v| *v != "with_prefix").map(|v| {
-                    syn::parse_str(v)
-                        .map_err(|e| syn::Error::new(s.span(), e))
-                        .expect_or_abort("invalid visibility found")
-                })
-            } else {
-                None
-            }
+// Helper function to extract string from Expr
+fn expr_to_string(expr: &Expr) -> Option<String> {
+    if let Expr::Lit(expr_lit) = expr {
+        if let Lit::Str(s) = &expr_lit.lit {
+            Some(s.value())
+        } else {
+            None
         }
-        _ => None,
+    } else {
+        None
+    }
+}
+
+// Helper function to parse visibility
+fn parse_vis_str(s: &str, span: proc_macro2::Span) -> Visibility {
+    match syn::parse_str(s) {
+        Ok(vis) => vis,
+        Err(e) => abort!(span, "Invalid visibility found: {}", e),
     }
 }
 
-/// Some users want legacy/compatability.
+// Helper function to parse visibility attribute
+pub fn parse_visibility(attr: Option<&Meta>, meta_name: &str) -> Option<Visibility> {
+    let meta = attr?;
+    let Meta::NameValue(MetaNameValue { value, path, .. }) = meta else {
+        return None;
+    };
+
+    if !path.is_ident(meta_name) {
+        return None;
+    }
+
+    let value_str = expr_to_string(value)?;
+    let vis_str = value_str.split(' ').find(|v| *v != "with_prefix")?;
+
+    Some(parse_vis_str(vis_str, value.span()))
+}
+
+/// Some users want legacy/compatibility.
 /// (Getters are often prefixed with `get_`)
 fn has_prefix_attr(f: &Field, params: &GenParams) -> bool {
-    let inner = f
-        .attrs
-        .iter()
-        .filter_map(|v| parse_attr(v, params.mode))
-        .filter(|meta| {
-            ["get", "get_copy"]
-                .iter()
-                .any(|ident| meta.path().is_ident(ident))
-        })
-        .last();
-
-    // Check it the attr includes `with_prefix`
-    let wants_prefix = |possible_meta: &Option<Meta>| -> bool {
-        match possible_meta {
-            Some(Meta::NameValue(meta)) => {
-                if let Lit::Str(lit_str) = &meta.lit {
-                    // Naive tokenization to avoid a possible visibility mod named `with_prefix`.
-                    lit_str.value().split(' ').any(|v| v == "with_prefix")
-                } else {
-                    false
+    // helper function to check if meta has `with_prefix` attribute
+    let meta_has_prefix = |meta: &Meta| -> bool {
+        if let Meta::NameValue(name_value) = meta {
+            if let Expr::Lit(expr_lit) = &name_value.value {
+                if let Lit::Str(s) = &expr_lit.lit {
+                    return s.value().split_whitespace().any(|v| v == "with_prefix");
```

Comment:
We should decide either on `split_whitespace()` or `split(' ')`. I think using both could cause confusion.